test/unit_tests/modules/FlyoutMenu-spec.js

```diff
@@ -0,0 +1,449 @@
+        'when called directly', function() {
+      flyoutMenu.collapse();
+    } );
+  } );
+
+  describe( '.setExpandTransition()', function() {
+    xit( 'should set a transition', function( done ) {
+      flyoutMenu.init();
+      var transition = new MoveTransition( contentDom );
+      flyoutMenu.setExpandTransition( transition, transition.moveLeft );
+      flyoutMenu.addEventListener( 'expandEnd', function() {
+        try {
+          var hasClass = contentDom.classList.contains( 'u-move-transition' );
+          expect( hasClass ).to.be.true;
+          done();
+        } catch( err ) {
+          done( err );
+        }
+      } );
+      triggerDom.click();
+    } );
+  } );
+
+  describe( '.setCollapseTransition()', function() {
+    it( 'should set a transition', function( done ) {
+      flyoutMenu.init();
+      var transition = new MoveTransition( contentDom );
+      triggerDom.click();
+      flyoutMenu.setCollapseTransition( transition, transition.moveLeft );
+      flyoutMenu.addEventListener( 'collapseEnd', function() {
+        try {
+          var hasClass = contentDom.classList.contains( 'u-move-transition' );
+          expect( hasClass ).to.be.true;
+          done();
+        } catch( err ) {
+          done( err );
+        }
+      } );
+      triggerDom.click();
+    } );
+  } );
+
+  describe( '.getTransition()', function() {
+    it( 'should return a transition instance', function() {
+      flyoutMenu.init();
+      expect( flyoutMenu.getTransition() ).to.be.undefined;
+      var transition = new MoveTransition( contentDom );
+      flyoutMenu.setExpandTransition( transition, transition.moveLeft );
+      flyoutMenu.setCollapseTransition( transition, transition.moveToOrigin );
+      expect( flyoutMenu.getTransition() ).to.equal( transition );
+      expect( flyoutMenu.getTransition( FlyoutMenu.COLLAPSE_TYPE ) )
+        .to.equal( transition );
+    } );
+  } );
+
+  describe( '.getDom()', function() {
+    it( 'should return references to full dom', function() {
+      flyoutMenu.init();
+      var dom = flyoutMenu.getDom();
+      expect( dom.container ).to.equal( containerDom );
+      expect( dom.trigger ).to.equal( triggerDom );
+      expect( dom.content ).to.equal( contentDom );
+      expect( dom.altTrigger ).to.equal( altTriggerDom );
+    } );
+  } );
+
+  describe( 'suspend/resume behavior', function() {
+    beforeEach( function() {
+      // Set up expected event listeners.
+      expandBeginSpy = sinon.spy();
+      expandEndSpy = sinon.spy();
+      collapseBeginSpy = sinon.spy();
+      collapseEndSpy = sinon.spy();
+      flyoutMenu.init();
+      flyoutMenu.addEventListener( 'expandBegin', expandBeginSpy );
+      flyoutMenu.addEventListener( 'expandEnd', expandEndSpy );
+      flyoutMenu.addEventListener( 'collapseBegin', collapseBeginSpy );
+      flyoutMenu.addEventListener( 'collapseEnd', collapseEndSpy );
+    } );
+
+    describe( '.resume()', function() {
```

Comment:
Since this depends on `suspend()` that test might want to come first, so if there's a failure, it's captured in the correct test in terminal output.